### PR TITLE
[AssetBundle] Move xml service configuration to php

### PR DESCRIPTION
**Pull request by NguyenTruongLinh**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Part of #37186
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -992,7 +992,7 @@ private function registerRequestConfiguration(array $config, ContainerBuilder $c
 
     private function registerAssetsConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:993]: Here we now need to change the type-hint for `$loader` to `PhpFileLoader` and ensure that in in `load()` we pass the `$phpLoader`.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+        ->set('assets.empty_package', Package::class)
+            ->args([
+                service('assets.empty_version_strategy')
+            ])
+
+        ->set('assets.context', RequestStackContext::class)
+            ->args([
+                service('request_stack'),
+                '%asset.request_context.base_path%',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:47]: `param('asset.request_context.base_path')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                '%asset.request_context.secure%'
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:48]: `param('asset.request_context.secure')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            ])
+
+        ->set('assets.path_package', PathPackage::class)
+            ->abstract()
+            ->args([
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:54]: `abstract_arg('base path')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:55]: `abstract_arg('version strategy')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+                service('assets.context')
+            ])
+
+        ->set('assets.url_package', UrlPackage::class)
+            ->abstract()
+            ->args([
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:62]: `abstract_arg('base URLs')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:63]: `abstract_arg('version strategy')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+                service('assets.context')
+            ])
+
+        ->set('assets.static_version_strategy', StaticVersionStrategy::class)
+            ->abstract()
+            ->args([
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:70]: `abstract_arg('version')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+                ''
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:71]: `abstract_arg('format')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+            ])
+
+        ->set('assets.empty_version_strategy', EmptyVersionStrategy::class)
+
+        ->set('assets.json_manifest_version_strategy', JsonManifestVersionStrategy::class)
+            ->abstract()
+            ->args([
+                ''
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:79]: `abstract_arg('manifest path')`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            ])
+
+        ->set('assets.remote_json_manifest_version_strategy', RemoteJsonManifestVersionStrategy::class)
+            ->abstract()
+            ->args([
+                '',
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.php:85]: `abstract_arg('manifest URL')`

**Review by xabbuh — CHANGES_REQUESTED** [review]
Thank you for working on this @NguyenTruongLinh. I have left some minor comments for a few things that need to be changed.

**Comment by NguyenTruongLinh** [comment]
> Thank you for working on this @NguyenTruongLinh. I have left some minor comments for a few things that need to be changed.

Thank you, I have updated it.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 12]
```diff
@@ -990,9 +990,9 @@ private function registerRequestConfiguration(array $config, ContainerBuilder $c
         }
     }
 
-    private function registerAssetsConfiguration(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    private function registerAssetsConfiguration(array $config, ContainerBuilder $container, PhpFileLoader $phpLoader)
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php:993]: please don't change the variable name here. and you need to call the function with the new `$phpLoader` variable on top

**Review by Tobion — CHANGES_REQUESTED** [review]

**Comment by Tobion** [comment]
Commas are missing: https://fabbot.io/report/symfony/symfony/37255/ede897ca97ec592202fccb223b8ea59170a8dd98

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 12 above)
Comment (NguyenTruongLinh) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Thank for response. I have updated

**Review by Tobion — CHANGES_REQUESTED** [review]
fabbot coding style checks are still failing

**Comment by NguyenTruongLinh** [comment]
> fabbot coding style checks are still failing

I have updated, but AppVeyor build test failed. What am I missing?

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Test failure is unrelated

**Comment by Tobion** [comment]
Thank you @NguyenTruongLinh.

**Pull request merged**